internal/pkg/runtime/engine/apptainer/process_linux.go

```diff
@@ -77,8 +77,18 @@ func (e *EngineOperations) StartProcess(masterConnFd int) error {
 	bootInstance := isInstance && e.EngineConfig.GetBootInstance()
 	shimProcess := false
 
+	customCwd := true
+	switch e.EngineConfig.OciConfig.Process.Cwd {
+	case e.EngineConfig.GetCwd(), e.EngineConfig.GetHomeDest():
+		customCwd = false
```

Comment:
I don't think this is a sufficient check to see if the `--pwd` option was used.  My guess is that there probably needs to be an additional parameter passed through the engineConfig when `cli.PwdPath` has been set.

Reply:
The only case where it would not report an error is with the current working directory being the same as the one specified with `--pwd`, so yes a boolean flag would be required in this case